qiskit/transpiler/target.py

```diff
@@ -1311,151 +1314,218 @@ def from_configuration(
             pulse_alignment = timing_constraints.pulse_alignment
             acquire_alignment = timing_constraints.acquire_alignment
 
-        qubit_properties = None
-        if backend_properties is not None:
-            # pylint: disable=cyclic-import
-            from qiskit.providers.backend_compat import qubit_props_list_from_props
-
-            qubit_properties = qubit_props_list_from_props(properties=backend_properties)
-
-        target = cls(
-            num_qubits=num_qubits,
-            dt=dt,
-            granularity=granularity,
-            min_length=min_length,
-            pulse_alignment=pulse_alignment,
-            acquire_alignment=acquire_alignment,
-            qubit_properties=qubit_properties,
-            concurrent_measurements=concurrent_measurements,
-        )
-        name_mapping = get_standard_gate_name_mapping()
+        required_operations = {"measure", "delay"}
```

Comment:
if 'delay' is removed here,
target does not have `delay`, because you see,
'delay' is only added when delay is present in 'required_operations'
'delay' is added into `inst_name_map` through `required_operations`
then, `delay` is added to `prop_name_map`
then finally, `delay` is added to `target` through `inst_name_map` and `prop_name_map`

To avoid all this hassle,
One, could just write something like this at the very end before `return target`
if in_data_target['num_qubits'] is not None:
    target.add_instruction(
    instruction=qiskit_inst_mapping.get('delay'),
    properties={(q,): None for q in range(num_qubits) if q not in faulty_qubits},
    name='delay'
    )

My opinion is, it's just a personal preference :)
Naoki , Kento, please correct me if I am missing something :)